api/envoy/config/route/v3/route_components.proto

```diff
@@ -443,7 +443,7 @@ message RouteMatch {
     ConnectMatcher connect_matcher = 12;
   }
 
-  // Indicates that prefix/path matching should be case insensitive. The default
+  // Indicates that prefix/path/safe_regex matching should be case insensitive. The default
```

Comment:
I'm a little concerned about the fact that this changes the behavior of an existing field instead of adding a new one, which could cause some interoperability issues.  No one will know whether the server intends to say that just the path or prefix should be matched case-insensitively or also the regexp.

In this particular case, I guess it's not terrible, because the regex and path/prefix matchers can't be specified together, which means that if `case_sensitive` is set when `safe_regex` is set, the meaning is unambiguous.  But the precedent is a little concerning to me.

As an alternative, would it make sense to add a `case_sensitive` field to `RegexMatcher`?  That would allow its behavior to be controlled independently.

I think the underlying question here is, does it make sense for `RegexMatcher` to have its behavior altered by its caller, or do we want to treat it like a black box that fully contains its own config?  If we say that its behavior can be altered by its caller, then that means that every caller will need to have its own way of determining whether the match should be case-sensitive, and the implementation of `RegexMatcher` will need to allow the caller to pass in a parameter to control this.  Conversely, if we say that it should be a black box, then it should not have this parameter, and it should have its own config field to control this.

Reply:
I could go either way on this one. At least on the Envoy side if we bundle this in, I would ask it to be runtime flag protected on a temporary basis to make sure we don't break anyone.

>  Conversely, if we say that it should be a black box, then it should not have this parameter, and it should have its own config field to control this.

I tend to agree that for regex it probably makes more sense for this to be owned by the regex config itself, especially since for regex it's not completely clear what case insensitive means. Does it mean that the input string is lower cases before the regex? Does it mean that both the input string and the regex itself are lower-cased? Etc. This probably is regex implementation dependent also.